File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Collaboration/CollaborationLog.vb
<Serializable()> _
Public Class CollaborationLog
    Inherits DomainObject(Of Integer)
    Implements ILog

#Region " Fields "

#End Region

#Region " Properties "

    Public Overrides Property Id As Integer Implements ILog.Id

    Public Overridable Property IdCollaboration As Integer

    Public Overridable Property CollaborationIncremental As Integer?

    Public Overridable Property Incremental As Short?

    Public Overridable Property IdChain As Integer?

    Public Overridable Property LogDate As Date Implements ILog.LogDate

    Public Overridable Property SystemComputer As String Implements ILog.SystemComputer

    Public Overridable Property SystemUser As String Implements ILog.SystemUser

    Public Overridable Property Program As String Implements ILog.Program

    Public Overridable Property LogType As String Implements ILog.LogType

    Public Overridable Property LogDescription As String Implements ILog.LogDescription

    Public Overridable Property Collaboration As Collaboration

    Public Overridable Property Number As Integer Implements ILog.Number

    Public Overridable Property Year As Short Implements ILog.Year

    Public Overridable Property Severity As Short? Implements ILog.Severity

#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Collaborations/CollaborationLog.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.Collaborations
{
    public class CollaborationLog : DSWBaseLogEntity<Collaboration, string>
    {
        #region [ Constructor ]

        public CollaborationLog() : this(Guid.NewGuid()) { }

        public CollaborationLog(Guid uniqueId)
            : base(uniqueId) { }
        #endregion

        #region [ Properties ]

        public DateTime? LogDate { get; set; }
        /// <summary>
        /// Get or set CollaborationIncremental
        /// </summary>
        public int? CollaborationIncremental { get; set; }

        /// <summary>
        /// Get or set Incremental
        /// </summary>
        public short? Incremental { get; set; }

        /// <summary>
        /// Get or set IdChain
        /// </summary>
        public int? IdChain { get; set; }

        /// <summary>
        /// Get or set Program
        /// </summary>
        public string Program { get; set; }

        #endregion

        #region [ Navigation Properties ]

        #endregion
    }
}
